src/tokens.js

```diff
@@ -93,16 +93,19 @@ const tokens = {
    *
    * @param {string} uid Token uid to be unregistered
    *
-   * @return {Array} array of token configs without the unregister one
+   * @return {Promise} promise that will be resolved with an array of tokens (after unregister) if succeds and the error in case of failure
    *
    * @memberof Tokens
    * @inner
    */
   unregisterToken(uid) {
-    const tokens = this.getTokens();
-    const filteredTokens = tokens.filter((token) => token.uid !== uid);
-    this.saveToStorage(filteredTokens);
-    return filteredTokens;
+    const promise = new Promise((resolve, _) => {
```

Comment:
I don't think we should use an exception here. There are no async operations.

Reply:
If the user decides to use an async storage (e.g. https://github.com/callstack/async-storage) it might have.

Reply:
I agree that nothing is prepared, we still need a big refactor. But if we start preparing every code we change it's better.

It's like this issue here: https://github.com/HathorNetwork/hathor-wallet-lib/issues/20. We still have lot of places that we need to change but the new ones (or when we change something), we are already updating.

To be honest the error handling is needed. This way or using a `try ... catch`. I don't see anything bad (it's a little bit slower but almost nothing because it's a simple code)

Reply:
In this case I don't think it makes sense because we're indicating the operation is async, when it's clearly not. In other operations where we only access the storage (like `tokenInfoExists()` or `getTokens()`), we don't use promises. This would make the lib more confusing to users.

Reply:
We are indicating that the operation may be async, which is true. We know that we need a larger refactor to allow async operations in the whole storage, but I feel new methods should do it from the beginning. I don't think it is more confusing for users because all classes have some sync and some async methods.

Reply:
We're indicating the operation *IS* async.

>The Promise object represents the eventual completion (or failure) of an asynchronous operation, and its resulting value.

We just need to make the standard in our lib then to always use promises instead of try/catch.